cmd/deploy/command_configuration.go

```diff
@@ -49,18 +64,18 @@ func setDeployOptionsValuesFromManifest(ctx context.Context, deployOptions *Opti
 	}
 
 	if deployOptions.Name == "" {
-		if deployOptions.Manifest.Name != "" {
-			deployOptions.Name = deployOptions.Manifest.Name
-		} else {
-			c, _, err := okteto.NewK8sClientProviderWithLogger(k8sLogger).Provide(okteto.GetContext().Cfg)
-			if err != nil {
-				return err
-			}
-			inferer := devenvironment.NewNameInferer(c)
-			deployOptions.Name = inferer.InferName(ctx, cwd, okteto.GetContext().Namespace, deployOptions.ManifestPathFlag)
-			deployOptions.Manifest.Name = deployOptions.Name
+		c, _, err := okteto.NewK8sClientProviderWithLogger(k8sLogger).Provide(okteto.GetContext().Cfg)
+		if err != nil {
+			return err
 		}
 
+		n := Namer{
+			Workdir:      cwd,
+			KubeClient:   c,
+			ManifestName: deployOptions.Manifest.Name,
+			ManifestPath: deployOptions.ManifestPathFlag,
+		}
+		deployOptions.Name = n.ResolveName(ctx)
```

Comment:
We should update also `deployOptions.Manifest.Name` as we were doing before in line 61.